src/components/SamlConfiguration/index.test.jsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import Index from './index';
+import LmsApiService from './../../data/services/LmsApiService';
+import { REQUIRED_DATA_FIELDS } from './SamlProviderDataForm';
+
+const formData = new FormData();
+REQUIRED_DATA_FIELDS.forEach(field => formData.append(field, 'testdata'));
+
+describe('<SamlConfiguration /> ', () => {
+  it('creating provider config states', () => {
+    const wrapper = shallow(<Index enterpriseId="testEnterpriseId" />);
+    const spyPostNewProviderConfig = jest.spyOn(LmsApiService, 'postNewProviderConfig');
+
+    const mockResponseData = { data: {} };
+    formData.forEach((value, key) => {
+      mockResponseData.data[key] = value;
+    });
+
+    spyPostNewProviderConfig.mockImplementation(() => mockResponseData);
+    wrapper.instance().createProviderConfig(formData, () => {
+      expect(wrapper.state().providerConfig).toEqual(mockResponseData.data);
+    });
+  });
+
+  it('updating provider config states', () => {
+    const wrapper = shallow(<Index enterpriseId="testEnterpriseId" />);
+    const spyPostUpdatedProviderConfig = jest.spyOn(LmsApiService, 'updateProviderConfig');
+
+    const oldProviderConfigData = { data: {} };
+    const mockResponseData = { data: {} };
+
+    formData.forEach((value, key) => {
+      oldProviderConfigData.data[key] = 'old '.concat(value);
+      mockResponseData.data[key] = value;
+    });
+    wrapper.setState({ providerConfig: oldProviderConfigData });
```

Comment:
Provider Config shouldn't be getting saved as a nested dict of `data` so using oldProviderConfigData here could introduce weird bugs or errors since we don't expect providerConfig to be structured as `{data: {field:value}}`

Reply:
I say "could introduce bugs" but I think that's a bit strong. We are rendering a saml provider config preview and a provider data preview which if we tested here (or would do in the future) the state being `{data: {}}` would be a problem.

Reply:
Agreed. Mix up with mock responses and what's set as the state. Will change back.